pkg/api/controller_test.go

```diff
@@ -2687,6 +2687,43 @@ func TestController_ObjectsGetObjectHandler(t *testing.T) {
 			t.Errorf("expected to get \"%s\" storage class, got %#v", expensiveString, properties)
 		}
 	})
+
+	t.Run("get object returns expected response with different etag", func(t *testing.T) {
+		newChecksum := `"11ee22ff33445566778899"`
+		eTagInput := "\"" + newChecksum + "\""
+		resp, err := clt.GetObjectWithResponse(ctx, repo, "main", &apigen.GetObjectParams{
+			Path:        "foo/bar",
+			IfNoneMatch: &eTagInput,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		if resp.HTTPResponse.StatusCode != http.StatusOK {
+			t.Errorf("GetObject() status code %d, expected %d", resp.HTTPResponse.StatusCode, http.StatusOK)
+		}
+
+		if resp.HTTPResponse.ContentLength != 37 {
+			t.Errorf("expected 37 bytes in content length, got back %d", resp.HTTPResponse.ContentLength)
+		}
+		etag := resp.HTTPResponse.Header.Get("ETag")
+		if etag != `"3c4838fe975c762ee97cf39fbbe566f1"` {
```

Comment:
I know this has been used like this before, but let's make this expected etag a const and use it wherever this string is used literally. This will help us make the tests more robust